esphome/components/as5600/sensor/as5600_sensor.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "esphome/core/component.h"
+#include "esphome/core/hal.h"
+#include "esphome/core/preferences.h"
+#include "esphome/components/i2c/i2c.h"
+#include "esphome/components/sensor/sensor.h"
+#include "esphome/components/as5600/as5600.h"
+
+namespace esphome {
+namespace as5600 {
+
+class AS5600Sensor : public PollingComponent, public Parented<AS5600Component>, public sensor::Sensor {
+ public:
+  AS5600Sensor();
+
+  void update() override;
+  void dump_config() override;
+  float get_setup_priority() const override;
+
+  void set_angle_sensor(sensor::Sensor *angle_sensor) { angle_sensor_ = angle_sensor; }
```

Comment:
Please make sure all member variables are prefixed with `this->` as per our [codebase standards](https://esphome.io/guides/contributing#codebase-standards)

```suggestion
  void set_angle_sensor(sensor::Sensor *angle_sensor) { this->angle_sensor_ = angle_sensor; }
```